### PR TITLE
Add support for vsomeip3 selinux policy

### PR DESCRIPTION
**Pull request by rhatdan**
This needs to wait until vsomeip3 selinux policy gets corrected and packaged.

### CODE REVIEW

**Comment by rhatdan** [comment]
@alexlarsson PTAL

**Comment by Yarboa** [comment]
> This needs to wait until vsomeip3 selinux policy gets corrected and packaged.

@rhatdan is it needed for that manifest? 
https://gitlab.com/CentOS/automotive/sample-images/-/merge_requests/377/diffs

**Comment by alexlarsson** [comment]
LGTM

**Comment by dougsland** [comment]
@rhatdan, CI is getting the output below:

```
+ export CXX
+ cd qm-0.5.8
+ /usr/bin/make all
mkdir -p tmp; cp qm.* tmp/
make -C tmp -f /usr/share/selinux/devel/Makefile qm.pp
make[1]: Entering directory '/builddir/build/BUILD/qm-0.5.8/tmp'
Compiling targeted qm module
qm.te:3:ERROR 'unknown type vsomeip_t' at token ';' on line 17149:
        allow qm_t vsomeip_t:unix_stream_socket connectto;
#line 3
/usr/bin/checkmodule:  error(s) encountered while parsing configuration
make[1]: *** [/usr/share/selinux/devel/include/Makefile:157: tmp/qm.mod] Error 1
make[1]: Leaving directory '/builddir/build/BUILD/qm-0.5.8/tmp'
make: *** [Makefile:28: qm.pp] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.21stYX (%build)
    Bad exit status from /var/tmp/rpm-tmp.21stYX (%build)

RPM build errors:
Finish: rpmbuild qm-0.5.8-1.20231026200407715281.pr244.51.gf255541.fc39.src.rpm
Finish: build phase for qm-0.5.8-1.20231026200407715281.pr244.51.gf255541.fc39.src.rpm
INFO: chroot_scan: 3 files copied to /var/lib/copr-rpmbuild/results/chroot_scan
INFO: /var/lib/mock/fedora-39-x86_64-1698350693.605672/root/var/log/dnf.rpm.log
/var/lib/mock/fedora-39-x86_64-1698350693.605672/root/var/log/dnf.librepo.log
/var/lib/mock/fedora-39-x86_64-1698350693.605672/root/var/log/dnf.log
ERROR: Exception(/var/lib/copr-rpmbuild/results/qm-0.5.8-1.20231026200407715281.pr244.51.gf255541.fc39.src.rpm) Config(fedora-39-x86_64) 0 minutes 34 seconds
INFO: Results and/or logs in: /var/lib/copr-rpmbuild/results
INFO: Cleaning up build root ('cleanup_on_failure=True')
Start: clean chroot
INFO: unmounting tmpfs.
Finish: clean chroot
ERROR: Command failed: 
 # /usr/bin/systemd-nspawn -q -M 1d5f341936b14e98a1b0d125f5139a83 -D /var/lib/mock/fedora-39-x86_64-1698350693.605672/root -a -u mockbuild --capability=cap_ipc_lock --rlimit=RLIMIT_NOFILE=10240 --capability=cap_ipc_lock --bind=/tmp/mock-resolv.g96fk4y9:/etc/resolv.conf --bind=/dev/btrfs-control --bind=/dev/mapper/control --bind=/dev/fuse --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin '--setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007"' '--setenv=PS1=<mock-chroot> \s-\v\$ ' --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c '/usr/bin/rpmbuild -bb  --target x86_64 --nodeps /builddir/build/originals/qm.spec'

Copr build error: Build failed
```

**Comment by rhatdan** [comment]
Smooge was supposed to be updating the vsomeip package to be written correctly.

**Comment by rhatdan** [comment]
vsomeip3-3.3.8-1.fc40

**Comment by dougsland** [comment]
still not flying, I will try next week a fresh CI/CD job.

**Comment by dougsland** [comment]
@rhatdan still failing, anything we should look?

**Comment by dougsland** [comment]
For some reason:

```
No matching package to install: 'vsomeip3-selinux >= 3.3.8'
Not all dependencies satisfied
```

investigating...

**Comment by dougsland** [comment]
@Yarboa  just a heads up, I also would like to include this one before the next release. We need to figure out why it's not working, seems not related to the patch but build?

**Comment by Yarboa** [comment]
> @Yarboa just a heads up, I also would like to include this one before the next release. We need to figure out why it's not working, seems not related to the patch but build?

Correct, this PR should be rebased inorder to pass tests it is uses qm-0.5.8

> @Yarboa just a heads up, I also would like to include this one before the next release. We need to figure out why it's not working, seems not related to the patch but build?

I assume spec file should install epel rpm
```
%if 0%{?rhel} && 0%{?centos} == 0
# Add EPEL repository for RHEL 9 CentOSStream
Requires: epel-release
%endif
Requires: vsomeip-selinux
```

**Comment by dougsland** [comment]
> %if 0%{?rhel} && 0%{?centos} == 0
> # Add EPEL repository for RHEL 9 CentOSStream
> Requires: epel-release
> %endif
> Requires: vsomeip-selinux

thanks, lets have a try ;-)

**Comment by dougsland** [comment]
holy guacamole, not sure if this is a real error, re-triggering the jobs.

```
[2024-01-09 06:15:30,449][ ERROR][PID:1455460] Re-try request for task on 'ResallocHost, ticket_id=568139, hostname=2620:52:3:1:dead:beef:cafe:c111': Unable to finish after 1 SSH attempts
[2024-01-09 06:15:30,449][  INFO][PID:1455460] Releasing VM back to pool
```

File: rpm/qm.spec [hunk 1]
```diff
@@ -59,6 +59,11 @@ BuildArch: noarch
 BuildRequires: golang-github-cpuguy83-md2man
 BuildRequires: container-selinux
 BuildRequires: make
+%if 0%{?rhel} && 0%{?centos} == 0
+# Add EPEL repository for RHEL 9 CentOSStream
+BuildRequires: epel-release
+%endif
+BuildRequires: vsomeip-selinux
```
Comment (Yarboa) [inline rpm/qm.spec:66]: @dougsland 
I think this is the error is should be vsomeip3-selinux

File: rpm/qm.spec [hunk 2]
```diff
@@ -59,6 +59,10 @@ BuildArch: noarch
 BuildRequires: golang-github-cpuguy83-md2man
 BuildRequires: container-selinux
 BuildRequires: make
+%if 0%{?rhel} && 0%{?centos} == 0
```
Comment (dougsland) [inline rpm/qm.spec:62]: I believe its: %if 0%{?centos} || 0%{?rhel}

**Comment by Yarboa** [comment]
> CI/CD error raised by pre-commit-checks not related and fixed in another PR:
> 
> ```
> demos/asilservice:25:1: B036 Don't except `BaseException` unless you plan to re-raise it.
> ```

Fixed in #311 
The build is related to some thing else,
see PR #298 

I need to rebase with #311 and see if solved

File: rpm/qm.spec [hunk 3]
```diff
@@ -59,6 +59,11 @@ BuildArch: noarch
 BuildRequires: golang-github-cpuguy83-md2man
 BuildRequires: container-selinux
 BuildRequires: make
+%if 0%{?centos} || 0%{?rhel}
+# Add EPEL repository for RHEL 9 CentOSStream
+BuildRequires: epel-release
```
Comment (lsm5) [inline rpm/qm.spec:64]: Probably won't help. Would be much easier to use `epel-` targets for copr as those can be used by TMT for rhel and centos tests. It's currently being changed in https://github.com/containers/qm/pull/298/files#diff-2b4300f6d69654fd71c0cdd329b5a5f13fa8fe7eb709c76276bdfd3ea1dde578R23 .

Comment (Yarboa) [reply rpm/qm.spec:64]: @dougsland please refer this #298 it is playground, at least to build the rpm

**Comment by dougsland** [comment]
Merged, we fix the packit later on the road. See #298

**Pull request merged**